### PR TITLE
SCF Assignment 2 initial commit

### PR DESCRIPTION
**Pull request by meta-aman-gautam**

### CODE REVIEW

File: Assignment_2/hexadecimalConversion/HexCalc.java [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+	 * 
+	 * @return {integer}value     Decimal equivalent of Hexadecimal no
+	 */
+	public int getDecimal(String hexnumber) {
+		int index, value = 0;
+		char character;
+		String digits = "0123456789ABCDEF", string;
+
+		hexnumber = hexnumber.toUpperCase();
+		for (int i = 0; i < hexnumber.length(); i++) {
```
Comment (meta-prateek-jain) [inline Assignment_2/hexadecimalConversion/HexCalc.java]: Variable name should be meaningful

File: Assignment_2/hexadecimalConversion/HexCalc.java [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+	 * Decimal no.
+	 * 
+	 * @param  {String}hexnumber   Hexadecimal no. user input
+	 * 
+	 * @return {integer}value     Decimal equivalent of Hexadecimal no
+	 */
+	public int getDecimal(String hexnumber) {
+		int index, value = 0;
+		char character;
+		String digits = "0123456789ABCDEF", string;
```
Comment (meta-prateek-jain) [inline Assignment_2/hexadecimalConversion/HexCalc.java]: Never use reserved keywords as a variable name

File: Assignment_2/hexadecimalConversion/HexCalc.java [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+
+		hexnumber = hexnumber.toUpperCase();
+		for (int i = 0; i < hexnumber.length(); i++) {
+			character = hexnumber.charAt(i);
+			string = Character.toString(character);
+			if (digits.contains(string)) {
+				index = digits.indexOf(character);
+				value = 16 * value + index;
+
```
Comment (meta-prateek-jain) [inline Assignment_2/hexadecimalConversion/HexCalc.java]: Empty Spaces

File: Assignment_2/hexadecimalConversion/HexCalc.java [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+	 * this is check method for checking if hexadecimal no. is valid or invalid
+	 * 
+	 * @param  {String} hexnumber
+	 * 
+	 * @return {boolean} value 
+	 */
+	public boolean check(String hexnumber) {
+		char character;
+		String string;
+		String digits = "0123456789ABCDEF";
```
Comment (meta-prateek-jain) [inline Assignment_2/hexadecimalConversion/HexCalc.java]: This variable can be defined as a constant

File: Assignment_2/hexadecimalConversion/HexCalc.java [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+			int flag = 0;
+			if (firstnumber.length() == secondnumber.length()) {
+
+				for (int count = 0; count < firstnumber.length(); count++) {
+
+					if (firstnumber.charAt(count) != secondnumber.charAt(count)) {
+						value = false;
+						break;
+					} else {
+						if (flag == firstnumber.length() - 1) {
```
Comment (meta-prateek-jain) [inline Assignment_2/hexadecimalConversion/HexCalc.java]: Code can be optimized

File: Assignment_2/hexadecimalConversion/HexCalc.java [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+	 * @param {integer} decimalnumber
+	 * 
+	 * @return {String} value
+	 */
+	public String getHex(int decimalnumber) {
+		int remainder;
+		String value = "";
+		
+		char digit[] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A',
+				'B', 'C', 'D', 'E', 'F' };
```
Comment (meta-prateek-jain) [inline Assignment_2/hexadecimalConversion/HexCalc.java]: Can be defined as a constant as `final char digit[] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' };`

Comment (meta-aman-gautam) [reply Assignment_2/hexadecimalConversion/HexCalc.java]: changed

File: Assignment_2/hexadecimalConversion/HexCalc.java (hunk 1 above)
Comment (meta-aman-gautam) [reply Assignment_2/hexadecimalConversion/HexCalc.java]: changed to a meaning full name

File: Assignment_2/hexadecimalConversion/HexCalc.java (hunk 2 above)
Comment (meta-aman-gautam) [reply Assignment_2/hexadecimalConversion/HexCalc.java]: changed to temp (string type)

File: Assignment_2/hexadecimalConversion/HexCalc.java (hunk 3 above)
Comment (meta-aman-gautam) [reply Assignment_2/hexadecimalConversion/HexCalc.java]: removed

File: Assignment_2/hexadecimalConversion/HexCalc.java (hunk 4 above)
Comment (meta-aman-gautam) [reply Assignment_2/hexadecimalConversion/HexCalc.java]: const DIGIT defined

**Pull request merged**